[linter issues] some deadcode are detected by tool gometalinter/deadcode

### Ⅰ. Issue Description

I used [deadcode](https://github.com/tsenart/deadcode) in [gometalinter](https://github.com/alecthomas/gometalinter) to check the code of PouchContainer project, then I found that there are some codes are unused:

```
ctrd/container.go:31:1:warning: defaultTrylockTimeout is unused (deadcode)
ctrd/utils.go:84:1:warning: generateID is unused (deadcode)
ctrd/utils.go:109:1:warning: ociImageToPouchImage is unused (deadcode)
daemon/containerio/hijack_conn.go:22:1:warning: createHijackConn is unused (deadcode)
daemon/daemon.go:56:1:warning: router is unused (deadcode)
daemon/daemon.go:120:1:warning: loadSymbolByName is unused (deadcode)
daemon/mgr/spec_linux.go:377:1:warning: isEmpty is unused (deadcode)
daemon/mgr/spec_linux.go:382:1:warning: isNone is unused (deadcode)
daemon/mgr/spec_linux.go:392:1:warning: isShareable is unused (deadcode)
daemon/mgr/spec_linux.go:403:1:warning: isPrivate is unused (deadcode)
main.go:262:1:warning: setLxcfsProcess is unused (deadcode)
pkg/utils/metrics/unit.go:7:1:warning: nanoseconds is unused (deadcode)
pkg/utils/metrics/unit.go:7:1:warning: bytes is unused (deadcode)
registry/auth.go:74:1:warning: genV1Endpoints is unused (deadcode)
storage/quota/grpquota.go:21:1:warning: grpQuotaType is unused (deadcode)
storage/quota/prjquota.go:21:1:warning: prjQuotaType is unused (deadcode)
test/util_api.go:34:1:warning: CreateBusyboxContainerOk is unused (deadcode)
test/util_api.go:68:1:warning: CreateBusybox125Container is unused (deadcode)
test/util_api.go:85:1:warning: StartContainerOk is unused (deadcode)
test/util_api.go:94:1:warning: StopContainerOk is unused (deadcode)
test/util_api.go:103:1:warning: CheckContainerStatus is unused (deadcode)
test/util_api.go:115:1:warning: CheckContainerRunning is unused (deadcode)
test/util_api.go:128:1:warning: DelContainerForceOk is unused (deadcode)
test/util_api.go:145:1:warning: PauseContainerOk is unused (deadcode)
test/util_api.go:154:1:warning: UnpauseContainerOk is unused (deadcode)
test/util_api.go:163:1:warning: DelContainerForceMultyTime is unused (deadcode)
test/util_api.go:191:1:warning: DelImageForceOk is unused (deadcode)
test/util_api.go:203:1:warning: CreateExecEchoOk is unused (deadcode)
test/util_api.go:226:1:warning: StartContainerExec is unused (deadcode)
test/util_api.go:239:1:warning: CreateVolumeOK is unused (deadcode)
test/util_api.go:256:1:warning: RemoveVolumeOK is unused (deadcode)
test/util_api.go:266:1:warning: DelNetworkOk is unused (deadcode)
test/util_api.go:275:1:warning: PullImage is unused (deadcode)
test/util_api.go:313:1:warning: GetMetric is unused (deadcode)
test/util_daemon.go:48:1:warning: StartDefaultDaemonDebug is unused (deadcode)
test/util_daemon.go:58:1:warning: StartDefaultDaemon is unused (deadcode)
test/util_daemon.go:68:1:warning: StartDaemonBareWithArgs is unused (deadcode)
test/util_daemon.go:75:1:warning: RestartDaemon is unused (deadcode)
test/util_daemon.go:81:1:warning: RunWithSpecifiedDaemon is unused (deadcode)
test/util_daemon.go:103:1:warning: CreateConfigFile is unused (deadcode)
test/utils.go:15:1:warning: helloworldImageOnlyRepoName is unused (deadcode)
test/utils.go:27:1:warning: testHubAddress is unused (deadcode)
test/utils.go:27:1:warning: testHubPasswd is unused (deadcode)
test/utils.go:27:1:warning: clientWrongCa is unused (deadcode)
test/utils.go:27:1:warning: testDaemonHTTPSAddr is unused (deadcode)
test/utils.go:27:1:warning: busyboxImage125ID is unused (deadcode)
test/utils.go:27:1:warning: testHubUser is unused (deadcode)
test/utils.go:62:1:warning: helpwantedForMissingCase is unused (deadcode)
test/utils.go:70:1:warning: SkipIfFalse is unused (deadcode)
test/utils.go:79:1:warning: IsTLSExist is unused (deadcode)
test/utils.go:102:1:warning: inspectFilter is unused (deadcode)
```

If these are indeed deadcode, I think we should try to eliminate them from this project. Since dead code will lead to increasing maintainace cost.


### Ⅱ. Describe what happened

deadcode exists somewhere in project.

### Ⅲ. Describe what you expected to happen
 no deadcode there.

### Ⅳ. How to reproduce it (as minimally and precisely as possible)

none

### Ⅴ. Anything else we need to know?
none


### Ⅵ. Environment:

- pouch version (use `pouch version`):
- OS (e.g. from /etc/os-release):
- Kernel (e.g. `uname -a`):
- Install tools:
- Others:

--- a/ctrd/container.go
+++ b/ctrd/container.go
@@ -32,8 +32,7 @@ import (
 )
 
 var (
-	runtimeRoot           = "/run"
-	defaultTrylockTimeout = 5 * time.Second
+	runtimeRoot = "/run"
 )
 
 type containerPack struct {

--- a/ctrd/utils.go
+++ b/ctrd/utils.go
@@ -3,14 +3,12 @@ package ctrd
 import (
 	"context"
 	"crypto/tls"
-	"encoding/json"
 	"net"
 	"net/http"
 	"time"
 
 	"github.com/alibaba/pouch/apis/types"
 	"github.com/alibaba/pouch/pkg/errtypes"
-	"github.com/alibaba/pouch/pkg/utils"
 
 	"github.com/containerd/containerd/containers"
 	"github.com/containerd/containerd/errdefs"
@@ -18,9 +16,8 @@ import (
 	"github.com/containerd/containerd/oci"
 	"github.com/containerd/containerd/remotes"
 	"github.com/containerd/containerd/remotes/docker"
-	"github.com/opencontainers/go-digest"
 	"github.com/opencontainers/image-spec/specs-go/v1"
-	specs "github.com/opencontainers/runtime-spec/specs-go"
+	"github.com/opencontainers/runtime-spec/specs-go"
 	"github.com/pkg/errors"
 )
 
@@ -80,19 +77,6 @@ func resolver(authConfig *types.AuthConfig) (remotes.Resolver, error) {
 	return docker.NewResolver(options), nil
 }
 
-// generateID generates image's ID by the SHA256 hash of its configuration JSON.
-func generateID(config *types.ImageInfo) (digest.Digest, error) {
-	var ID digest.Digest
-
-	b, err := json.Marshal(config)
-	if err != nil {
-		return ID, err
-	}
-
-	ID = digest.FromBytes(b)
-	return ID, nil
-}
-
 // rootFSToAPIType transfer the rootfs from OCI format to Pouch format.
 func rootFSToAPIType(rootFs *v1.RootFS) types.ImageInfoRootFS {
 	var layers []string
@@ -103,39 +87,6 @@ func rootFSToAPIType(rootFs *v1.RootFS) types.ImageInfoRootFS {
 		Type:   rootFs.Type,
 		Layers: layers,
 	}
-}
-
-// ociImageToPouchImage transfer the image from OCI format to Pouch format.
-func ociImageToPouchImage(ociImage v1.Image) (types.ImageInfo, error) {
-	imageConfig := ociImage.Config
-
-	volumes := make(map[string]interface{})
-	for k, obj := range imageConfig.Volumes {
-		volumes[k] = obj
-	}
-	cfg := &types.ContainerConfig{
-		// TODO: add more fields
-		User:       imageConfig.User,
-		Env:        imageConfig.Env,
-		Entrypoint: imageConfig.Entrypoint,
-		Cmd:        imageConfig.Cmd,
-		WorkingDir: imageConfig.WorkingDir,
-		Labels:     imageConfig.Labels,
-		StopSignal: imageConfig.StopSignal,
-		Volumes:    volumes,
-	}
-
-	rootFs := rootFSToAPIType(&ociImage.RootFS)
-
-	// FIXME need to refactor it and the ociImage's list interface.
-	imageInfo := types.ImageInfo{
-		Architecture: ociImage.Architecture,
-		Config:       cfg,
-		CreatedAt:    ociImage.Created.Format(utils.TimeLayout),
-		Os:           ociImage.OS,
-		RootFS:       &rootFs,
-	}
-	return imageInfo, nil
 }
 
 // toLinuxResources transfers Pouch Resources to LinuxResources.

--- a/daemon/daemon.go
+++ b/daemon/daemon.go
@@ -5,7 +5,6 @@ import (
 	"fmt"
 	"path"
 	"path/filepath"
-	"plugin"
 	"reflect"
 
 	"github.com/alibaba/pouch/apis/server"
@@ -24,8 +23,6 @@ import (
 
 	systemddaemon "github.com/coreos/go-systemd/daemon"
 	systemdutil "github.com/coreos/go-systemd/util"
-	"github.com/gorilla/mux"
-	"github.com/pkg/errors"
 	"github.com/sirupsen/logrus"
 )
 
@@ -50,12 +47,6 @@ type Daemon struct {
 	volumePlugin    hookplugins.VolumePlugin
 	criPlugin       hookplugins.CriPlugin
 	eventsService   *events.Events
-}
-
-// router represents the router of daemon.
-type router struct {
-	daemon *Daemon
-	*mux.Router
 }
 
 // NewDaemon constructs a brand new server.
@@ -115,14 +106,6 @@ func NewDaemon(cfg *config.Config) *Daemon {
 		ctrdDaemon:     ctrdDaemon,
 		containerStore: containerStore,
 	}
-}
-
-func loadSymbolByName(p *plugin.Plugin, name string) (plugin.Symbol, error) {
-	s, err := p.Lookup(name)
-	if err != nil {
-		return nil, errors.Wrapf(err, "lookup plugin with name %s error", name)
-	}
-	return s, nil
 }
 
 func (d *Daemon) loadPlugin() error {

--- a/daemon/mgr/spec_linux.go
+++ b/daemon/mgr/spec_linux.go
@@ -373,43 +373,15 @@ func setupNamespaces(ctx context.Context, c *Container, specWrapper *SpecWrapper
 	return setupUtsNamespace(ctx, c, specWrapper)
 }
 
-// isEmpty indicates whether namespace mode is empty.
-func isEmpty(mode string) bool {
-	return mode == ""
-}
-
-// isNone indicates whether container's namespace mode is set to "none".
-func isNone(mode string) bool {
-	return mode == "none"
-}
-
 // isHost indicates whether the container shares the host's corresponding namespace.
 func isHost(mode string) bool {
 	return mode == "host"
-}
-
-// isShareable indicates whether the containers namespace can be shared with another container.
-func isShareable(mode string) bool {
-	return mode == "shareable"
 }
 
 // isContainer indicates whether the container uses another container's corresponding namespace.
 func isContainer(mode string) bool {
 	parts := strings.SplitN(mode, ":", 2)
 	return len(parts) > 1 && parts[0] == "container"
-}
-
-// isPrivate indicates whether the container uses its own namespace.
-func isPrivate(ns specs.LinuxNamespaceType, mode string) bool {
-	switch ns {
-	case specs.IPCNamespace:
-		return mode == "private"
-	case specs.NetworkNamespace, specs.PIDNamespace:
-		return !(isHost(mode) || isContainer(mode))
-	case specs.UserNamespace, specs.UTSNamespace:
-		return !(isHost(mode))
-	}
-	return false
 }
 
 // connectedContainer is the id or name of the container whose namespace this container share with.

--- a/main.go
+++ b/main.go
@@ -17,7 +17,6 @@ import (
 	"github.com/alibaba/pouch/daemon/config"
 	"github.com/alibaba/pouch/lxcfs"
 	"github.com/alibaba/pouch/pkg/debug"
-	"github.com/alibaba/pouch/pkg/exec"
 	"github.com/alibaba/pouch/pkg/utils"
 	"github.com/alibaba/pouch/storage/quota"
 	"github.com/alibaba/pouch/version"
@@ -256,23 +255,6 @@ func initLog() {
 		TimestampFormat: time.RFC3339Nano,
 	}
 	logrus.SetFormatter(formatter)
-}
-
-// define lxcfs processe.
-func setLxcfsProcess(processes exec.Processes) exec.Processes {
-	if !cfg.IsLxcfsEnabled {
-		return processes
-	}
-
-	p := &exec.Process{
-		Path: cfg.LxcfsBinPath,
-		Args: []string{
-			cfg.LxcfsHome,
-		},
-	}
-	processes = append(processes, p)
-
-	return processes
 }
 
 // check lxcfs config

--- a/pkg/utils/metrics/unit.go
+++ b/pkg/utils/metrics/unit.go
@@ -5,8 +5,6 @@ package metrics
 type Unit string
 
 const (
-	nanoseconds Unit = "nanoseconds"
-	seconds     Unit = "seconds"
-	bytes       Unit = "bytes"
-	total       Unit = "total"
+	seconds Unit = "seconds"
+	total   Unit = "total"
 )

--- a/registry/auth.go
+++ b/registry/auth.go
@@ -70,11 +70,6 @@ func genRegistryEndpoints(addr string) (endpoints []registryEndpoint) {
 	return endpoints
 }
 
-// TODO: generate v2 registry endpoint for login
-func genV1Endpoints(addr string) registryEndpoint {
-	return registryEndpoint{}
-}
-
 func genV2Endpoints(addr string) registryEndpoint {
 	if addr == "" {
 		addr = defaultV2Registry

--- a/storage/quota/grpquota.go
+++ b/storage/quota/grpquota.go
@@ -18,10 +18,6 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	grpQuotaType = "grpquota"
-)
-
 // GrpQuotaDriver represents group quota driver.
 type GrpQuotaDriver struct {
 	lock sync.Mutex

--- a/storage/quota/prjquota.go
+++ b/storage/quota/prjquota.go
@@ -18,10 +18,6 @@ import (
 	"github.com/sirupsen/logrus"
 )
 
-var (
-	prjQuotaType = "prjquota"
-)
-
 // PrjQuotaDriver represents project quota driver.
 type PrjQuotaDriver struct {
 	lock sync.Mutex

--- a/test/util_api.go
+++ b/test/util_api.go
@@ -124,15 +124,6 @@ func CheckContainerRunning(c *check.C, cname string, isRunning bool) {
 	c.Assert(gotRunning, check.Equals, isRunning)
 }
 
-// DelContainerForceOk forcely deletes the container and asserts success.
-func DelContainerForceOk(c *check.C, cname string) {
-	resp, err := delContainerForce(cname)
-	c.Assert(err, check.IsNil)
-
-	defer resp.Body.Close()
-	CheckRespStatus(c, resp, 204)
-}
-
 func delContainerForce(cname string) (*http.Response, error) {
 	q := url.Values{}
 	q.Add("force", "true")
@@ -154,7 +145,6 @@ func PauseContainerOk(c *check.C, cname string) {
 func UnpauseContainerOk(c *check.C, cname string) {
 	resp, err := request.Post("/containers/" + cname + "/unpause")
 	c.Assert(err, check.IsNil)
-
 	defer resp.Body.Close()
 	CheckRespStatus(c, resp, 204)
 }

--- a/test/util_daemon.go
+++ b/test/util_daemon.go
@@ -64,13 +64,6 @@ func StartDefaultDaemon(args ...string) (*daemon.Config, error) {
 	return &cfg, cfg.StartDaemon()
 }
 
-// StartDaemonBareWithArgs starts a deamon with all user specified parameter.
-func StartDaemonBareWithArgs(cfg *daemon.Config, args ...string) error {
-	cfg.NewArgs(args...)
-
-	return cfg.StartDaemon()
-}
-
 // RestartDaemon restart daemon
 func RestartDaemon(cfg *daemon.Config) error {
 	cfg.KillDaemon()

